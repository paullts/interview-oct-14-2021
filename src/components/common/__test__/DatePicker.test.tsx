import React from 'react';
import { cleanup, render, screen } from '@testing-library/react';
import CustomDateRangePicker from '../DatePicker';

describe('renders <Table /> with header and empty rows', () => {
  let renderResult: any;
  const handleChange = jest.fn();
  beforeEach(() => {
    renderResult = render(
      <CustomDateRangePicker
        name='test-daterange-picker'
        date_from='2021-10-01T00:00:00'
        date_to='2021-10-03T23:59:59'
        handleChange={handleChange}
      />
    );
  });
  afterEach(cleanup);

  it('should match the snapshot', () => {
    expect(renderResult.container).toMatchSnapshot();
  });

  it('should display `10/01/2021` and `10/03/2021`', () => {
    expect(screen.getByDisplayValue(/2021-10-01/i)).toBeTruthy();
    expect(screen.getByDisplayValue(/2021-10-03/i)).toBeTruthy();
  });
});
