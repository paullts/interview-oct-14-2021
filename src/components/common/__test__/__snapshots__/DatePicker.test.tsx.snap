// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <Table /> with header and empty rows should match the snapshot 1`] = `
<div>
  <div
    class="react-daterange-picker react-daterange-picker--closed react-daterange-picker--enabled"
  >
    <div
      class="react-daterange-picker__wrapper"
    >
      <div
        class="react-daterange-picker__inputGroup"
      >
        <input
          max="275760-09-13"
          min="0001-01-01"
          name="test-daterange-picker_from"
          style="visibility: hidden; position: absolute; z-index: -999;"
          type="date"
          value="2021-10-01"
        />
        
        <input
          autocomplete="off"
          class="react-daterange-picker__inputGroup__input react-daterange-picker__inputGroup__month"
          data-input="true"
          inputmode="numeric"
          max="12"
          min="1"
          name="month"
          placeholder="--"
          type="number"
          value="10"
        />
        <span
          class="react-date-picker__inputGroup__divider"
        >
          /
        </span>
        <span
          class="react-daterange-picker__inputGroup__leadingZero"
        >
          0
        </span>
        <input
          autocomplete="off"
          class="react-daterange-picker__inputGroup__input react-daterange-picker__inputGroup__day react-daterange-picker__inputGroup__input--hasLeadingZero"
          data-input="true"
          inputmode="numeric"
          max="31"
          min="1"
          name="day"
          placeholder="--"
          type="number"
          value="1"
        />
        <span
          class="react-date-picker__inputGroup__divider"
        >
          /
        </span>
        <input
          autocomplete="off"
          class="react-daterange-picker__inputGroup__input react-daterange-picker__inputGroup__year"
          data-input="true"
          inputmode="numeric"
          max="275760"
          min="1"
          name="year"
          placeholder="----"
          step="1"
          type="number"
          value="2021"
        />
        
      </div>
      <span
        class="react-daterange-picker__range-divider"
      >
        â€“
      </span>
      <div
        class="react-daterange-picker__inputGroup"
      >
        <input
          max="275760-09-13"
          min="0001-01-01"
          name="test-daterange-picker_to"
          style="visibility: hidden; position: absolute; z-index: -999;"
          type="date"
          value="2021-10-03"
        />
        
        <input
          autocomplete="off"
          class="react-daterange-picker__inputGroup__input react-daterange-picker__inputGroup__month"
          data-input="true"
          inputmode="numeric"
          max="12"
          min="1"
          name="month"
          placeholder="--"
          type="number"
          value="10"
        />
        <span
          class="react-date-picker__inputGroup__divider"
        >
          /
        </span>
        <span
          class="react-daterange-picker__inputGroup__leadingZero"
        >
          0
        </span>
        <input
          autocomplete="off"
          class="react-daterange-picker__inputGroup__input react-daterange-picker__inputGroup__day react-daterange-picker__inputGroup__input--hasLeadingZero"
          data-input="true"
          inputmode="numeric"
          max="31"
          min="1"
          name="day"
          placeholder="--"
          type="number"
          value="3"
        />
        <span
          class="react-date-picker__inputGroup__divider"
        >
          /
        </span>
        <input
          autocomplete="off"
          class="react-daterange-picker__inputGroup__input react-daterange-picker__inputGroup__year"
          data-input="true"
          inputmode="numeric"
          max="275760"
          min="1"
          name="year"
          placeholder="----"
          step="1"
          type="number"
          value="2021"
        />
        
      </div>
      <button
        aria-label="Clear value"
        class="react-daterange-picker__clear-button react-daterange-picker__button"
        type="button"
      >
        <svg
          class="react-daterange-picker__clear-button__icon react-daterange-picker__button__icon"
          height="19"
          stroke="black"
          stroke-width="2"
          viewBox="0 0 19 19"
          width="19"
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            x1="4"
            x2="15"
            y1="4"
            y2="15"
          />
          <line
            x1="15"
            x2="4"
            y1="4"
            y2="15"
          />
        </svg>
      </button>
      <button
        aria-label="Toggle calendar"
        class="react-daterange-picker__calendar-button react-daterange-picker__button"
        type="button"
      >
        <svg
          class="react-daterange-picker__calendar-button__icon react-daterange-picker__button__icon"
          height="19"
          stroke="black"
          stroke-width="2"
          viewBox="0 0 19 19"
          width="19"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            fill="none"
            height="15"
            width="15"
            x="2"
            y="2"
          />
          <line
            x1="6"
            x2="6"
            y1="0"
            y2="4"
          />
          <line
            x1="13"
            x2="13"
            y1="0"
            y2="4"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
